nexus_constructor/model/component.py

```diff
@@ -84,10 +84,12 @@ class Component(Group):
     Base class for a component object. In the NeXus file this would translate to the component group.
     """
 
-    _depends_on = attr.ib(type=Transformation, default=None)
-    has_link = attr.ib(type=bool, default=None)
-    component_info: "ComponentInfo" = None
-    stored_transforms: list = None
+    _depends_on: Transformation = attr.ib(default=None)
+    has_link: bool = attr.ib(default=None)
+    component_info: Optional["ComponentInfo"] = None
+    stored_transforms: list = []
```

Comment:
I am trying to resist too much overly specific typing here, since we should still take advantage of Python's interpretation abilities.  The suggested change would result in the static checkers complaining about #98: since children have a different type, but as they are all 'list' it seems to supress the warnings here without causing others to occur.